src/physics/euler.cpp

```diff
@@ -569,6 +611,77 @@ ::convective_surface_numerical_split_flux (
     return flux_interp_to_surface;
 }
 
+template <int dim, int nstate, typename real>
+std::array<real,nstate> Euler<dim, nstate, real>
+::compute_entropy_variables (
+    const std::array<real,nstate> &conservative_soln) const
+{
+    std::array<real,nstate> entropy_var;
+    const real density = conservative_soln[0];
+    const real pressure = compute_pressure<real>(conservative_soln);
+    const dealii::Tensor<1,dim,real> vel = compute_velocities<real>(conservative_soln);
+    const real vel2 = compute_velocity_squared<real>(vel);
+    const real entropy = log(pressure) - gam * log(density);
+    const real rho_theta = conservative_soln[nstate-1] - 0.5 * density * vel2;
+
+   // entropy_var[0] = (gam-entropy)/gamm1 - 0.5 * density / pressure * vel2;
+    entropy_var[0] = (rho_theta *(gam + 1.0 - entropy) - conservative_soln[nstate-1])/rho_theta;
+//    std::cout<<"entropy var 1 first "<<entropy_var[0]<<std::endl;
+    for(int idim=0; idim<dim; idim++){
+       // entropy_var[idim+1] = conservative_soln[idim+1] / pressure;
+        entropy_var[idim+1] = conservative_soln[idim+1] / rho_theta;
+//    std::cout<<"entropy var 2 first "<<entropy_var[1]<<std::endl;
+    }
+   // entropy_var[nstate-1] = - density / pressure;
+    entropy_var[nstate-1] = - density / rho_theta;
+//    std::cout<<"entropy var 3 first "<<entropy_var[2]<<std::endl;
+
+//    const real rhoe = conservative_soln[nstate-1] - 0.5*density*vel2;
+//    entropy_var[0] = (conservative_soln[nstate-1] - rhoe * ( gam + 1 - entropy))/rhoe;
+//    std::cout<<"entropy var 1 second "<<entropy_var[0]<<std::endl;
+//    for(int idim=0; idim<dim; idim++){
+//        entropy_var[idim+1] = conservative_soln[idim+1] / rhoe;
+//    std::cout<<"entropy var 2 second "<<entropy_var[1]<<std::endl;
+//    }
+//    entropy_var[nstate-1] = - density / rhoe;
+//    std::cout<<"entropy var 3 second "<<entropy_var[2]<<std::endl;
+
+    //kinetic energy entropy variables
+//    entropy_var[0] = 0.0;
+//    for(int idim=0; idim<dim; idim++){
+//        entropy_var[0] += 0.5 * vel[idim] * vel[idim];
+//        entropy_var[idim+1] = vel[idim];
+//    }
+//   // entropy_var[nstate-1] = dim + 1;
+//    entropy_var[nstate-1] = 0.0;
```

Comment:
can also remove the commented code from here

Reply:
Fixed.